File: packages/dvai-bridge-android-shared-core/android/src/main/java/co/deepvoiceai/bridge/shared/core/offload/OffloadConfig.kt
package co.deepvoiceai.bridge.shared.core.offload

import co.deepvoiceai.bridge.shared.core.discovery.Peer

/**
 * Configuration for distributed inference / device offload (Kotlin
 * mirror of the TS `OffloadConfig` in
 * `@westenets/dvai-bridge-core/src/offload/types.ts`).
 *
 * Offload is opt-in: pass an `OffloadConfig(enabled = true)` to
 * `StartOptions.offload` to turn it on. Default is null = behave
 * exactly like v2.x.
 *
 * @param enabled              Master switch. Default false.
 * @param discoverLAN          Run mDNS / NsdManager discovery for LAN peers. Default true.
 * @param minLocalCapability   Below this tok/s, look for a peer. Default 10.0.
 * @param hardwareMinimum      v3.2 — hard floor for any local inference, in tok/s.
 *                             Below this, the device is "too weak"; the SDK
 *                             aborts start() and (optionally) calls
 *                             [onHardwareTooWeak] so the host can show a
 *                             system popup. Default 3.0.
 * @param onHardwareTooWeak    v3.2 — host hook fired when the precheck
 *                             classifies the device as too-weak. Use to surface
 *                             a platform-native AlertDialog. The SDK still
 *                             throws afterward so start() fails in a
 *                             structured way (DVAIBridgeError.HardwareTooWeak).
 * @param rendezvousUrl        Optional rendezvous-server URL — enables internet path.
 * @param knownPeers           Optional pre-known peers (skip discovery for them).
 * @param onPairingRequest     Suspending callback returning approve/deny for incoming
 *                             pairings. If null, [PairingPolicy.requests] Flow is the
 *                             only path; if both are absent, default-deny applies.
 *                             Mirrors the JS callback shape.
 * @param onOffload            Diagnostic callback when a request is offloaded. Optional.
 */
data class OffloadConfig(
    val enabled: Boolean = false,
    val discoverLAN: Boolean = true,
    val minLocalCapability: Double = 10.0,
    val hardwareMinimum: Double = 3.0,
    val onHardwareTooWeak: ((HardwareTooWeakInfo) -> Unit)? = null,
    val rendezvousUrl: String? = null,
    val knownPeers: List<Peer> = emptyList(),
    val onPairingRequest: (suspend (Peer) -> Boolean)? = null,
    val onOffload: ((Peer) -> Unit)? = null,
)

/** Detail surfaced to [OffloadConfig.onHardwareTooWeak]. */
data class HardwareTooWeakInfo(
    val tokPerSec: Double,
    val hardwareMinimum: Double,
    val reason: String,
)

File: packages/dvai-bridge-dotnet/src/DVAIBridge/Offload/OffloadConfig.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace DVAIBridge;

/// <summary>
/// Distributed-inference (offload) configuration. Optionally attached to
/// <see cref="StartOptions.Offload"/>. When <see cref="Enabled"/> is
/// <c>true</c>, the bridge runs LAN discovery (mDNS) for sibling devices
/// and surfaces a <see cref="DVAIBridge.PairingRequests"/> stream for
/// host-app UI. Mirrors the TypeScript <c>OffloadConfig</c> in
/// <c>@dvai-bridge/core</c>.
/// </summary>
public sealed record OffloadConfig
{
    /// <summary>Master switch. Default <c>false</c>; offload is opt-in at v3.0.</summary>
    public bool Enabled { get; init; } = false;

    /// <summary>Run mDNS to discover LAN peers. Default <c>true</c>.</summary>
    public bool DiscoverLAN { get; init; } = true;

    /// <summary>
    /// Below this tok/s estimate of local capability, the offload decider
    /// looks for a peer. Default <c>10</c> tok/s.
    /// </summary>
    public double MinLocalCapability { get; init; } = 10.0;

    /// <summary>
    /// Optional rendezvous-server URL — enables the internet path when set.
    /// Should be a <c>wss://</c> or <c>https://</c> URL pointing at a
    /// self-hosted rendezvous deployment.
    /// </summary>
    public Uri? RendezvousUrl { get; init; }

    /// <summary>
    /// Optional pre-known peers (skip discovery for these). Useful for
    /// kiosk / fleet deployments where the peer addresses are known in
    /// advance.
    /// </summary>
    public IReadOnlyList<PeerInfo> KnownPeers { get; init; } = Array.Empty<PeerInfo>();

    /// <summary>
    /// Hook to surface a pairing-request UI to the host app. Returning
    /// <c>true</c> approves the pairing; <c>false</c> denies. Default:
    /// deny (no callback supplied). On desktop, prefer the streaming
    /// <see cref="DVAIBridge.PairingRequests"/> surface; mobile callers
    /// commonly use this callback because they don't host the UI thread
    /// the way desktop apps do.
    /// </summary>
    public Func<PeerInfo, Task<bool>>? OnPairingRequest { get; init; }
}
